### PR TITLE
Bump requirements files && codecov action to 1.5.2

### PR DESCRIPTION
**Pull request by eapolinario**
# TL;DR
Bump requirements and version of codecov-action

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
The codecov github action version 1.5.2 fixed a bug that shows up in flytekit PRs very frequently. This PR bumps requirements across the board and also also the version of the codecov github action.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Review by wild-endeavor — APPROVED** [review]

**Comment by wild-endeavor** [comment]
great expectations still failing but we can revert later right?  back to 0.15.6
